src/Runtime/Runtime/System.Windows.Controls/Image.cs

```diff
@@ -522,6 +522,15 @@ public override object CreateDomElement(object parentRef, out object domElementW
             style.height = "0"; // Same as above.
             style.objectPosition = "center top";
 
+            if (this.MaxHeight > 0)
```

Comment:
Do you think we should check for Double.IsNan or something like that?

Reply:
Yes, we should. Changing commit